packages/test/test-utils/src/testContainerRuntimeFactory.ts

```diff
@@ -54,32 +54,52 @@ export const createTestContainerRuntimeFactory = (
 		public async instantiateFromExisting(runtime: ContainerRuntime): Promise<void> {
 			// Validate we can load root data stores.
 			// We should be able to load any data store that was created in initializeFirstTime!
-			await runtime.getAliasedDataStoreEntryPoint("default");
+			// Note: This check was re-enabled to allow for cross-major version compat testing.
+			await (runtime.getAliasedDataStoreEntryPoint?.("default") ??
+				runtime.getRootDataStore("default"));
 		}
```

Comment:
1. `getRootDataStore` is deprecated in latest. Won't this make it harder to remove?
2. I see the convenience of using `?.(...) ??` to alternately call `getRootDataStore` but it isn't obvious at first. The runtime check may look at the return of `getAliasedDataStoreEntryPoint` but that isn't what is intended.
It would be nice to structure code for exactly what is desired even if it is a little more code.
3. Nit - comment the code for what is there rather than what change you are making. The test of "was re-enabled" means you have to know the history.

I haven't read enough to know how this all goes together to get older clients, but it seems like the typing is wrong.?

Reply:
> 1. `getRootDataStore` is deprecated in latest. Won't this make it harder to remove?
> 2. I see the convenience of using `?.(...) ??` to alternately call `getRootDataStore` but it isn't obvious at first. The runtime check may look at the return of `getAliasedDataStoreEntryPoint` but that isn't what is intended.
>    It would be nice to structure code for exactly what is desired even if it is a little more code.
> 3. Nit - comment the code for what is there rather than what change you are making. The test of "was re-enabled" means you have to know the history.
> 
> I haven't read enough to know how this all goes together to get older clients, but it seems like the typing is wrong.?

This is more of a temporary workaround to get the tests up and running again. We do have some similar cases in our test infra such as [waitContainerToCatchUp()](https://github.com/microsoft/FluidFramework/blob/82651ab38709f10ede0ae00066836a63c0f64bf9/packages/test/test-utils/src/testObjectProvider.ts#L470C1-L471C1). I think eventually we will have to handle this problem similarly when the latest version now longer has `getRootDataStore`. in that case, we will probably have to type cast to `any`, just like we do with `(container as any).resume` in `waitContainerToCatchUp`

Reply:
Not convinced you'd need to type cast to `any`. Seems like should be able to get the right typing in place. From what I can tell (plenty more to read), other versions are loaded but code pretends they are typed like current. That seems fraught.
If this code is temporary, please also add comments explaining.

Reply:
To clarify - We don't need to type cast to `any` now, but we may need to in the future when `getRootDataStore` is completely removed from the latest version.

Reply:
Will also clarify :), I am not convinced of future need. In general, if you need a cast, then something isn't great and perhaps not sound. `as` should not be used lightly IMO but there are some uses.